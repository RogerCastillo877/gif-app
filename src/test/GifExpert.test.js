import React from "react";
import { shallow } from "enzyme";
import { GifExpertApp } from '../GifExpertApp';

describe('should test GifExpert component', () => {
    
    test('should show the component properly', () => {
        
        const wrapper = shallow(<GifExpertApp />);

        expect( wrapper ).toMatchSnapshot();
    })
    
    test('should show a list of category', () => {
        
        const categories = ['samurai x', 'Dragon Ball']
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
    
})
