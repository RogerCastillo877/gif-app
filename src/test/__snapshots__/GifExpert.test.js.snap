// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test GifExpert component should show a list of category 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="samurai x"
      key="samurai x"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;

exports[`should test GifExpert component should show the component properly 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
